carla/models/catalog/catalog.py

```diff
@@ -313,3 +335,14 @@ def use_pipeline(self, use_pipe: bool) -> None:
 
         """
         self._use_pipeline = use_pipe
+
+    def train(self, learning_rate, epochs, batch_size):
```

Comment:
Just for my understanding @MartinPawel , @JohanvandenHeuvel  and @Philoso-Fish , did we align on adding training scripts to the library? In my mental model of the library the training and model handling should happen outside of the library. 

Users should still be able to train a model themselves and be easy for them to do recourse with them, but i think we introduct training in the core part of the library we are coupleing things, that shouldn't be coupled

Reply:
The main reason I started implementing now was because we want to add causal recourse methods. These kind of methods often use synthetic data, that are generated using Structural Causal Equations (SCM). Then we need models that are trained on these data. 

I guess this also violates your mental model as the data generated by the SCM does not live in the data repository. To be honest I didn't really think about the decoupling of recourse, data, and mlmodel. I'm not sure yet if we can decouple it for the causal recourse methods.

Reply:
@JohanvandenHeuvel ah yes i remember the disscussion around causal models. Is there a straightforward way to co-locate the model training with the causal models, or make it clear its for synthethic data only? 

Also to further clarify, since the causal models or using synthethic data, they are not usable in practical use caess?

Reply:
A synthetic dataset has the same structure as a "normal" dataset. So by adding model training for the synthetic dataset, it also works automatically for our other datasets. It doesn't make sense to me to make it less general. We would have to make the synthetic dataset less general on purpose, and then on purpose only support training of synthetic datasets.

I don't know enough about the topic to comment on the second question.

Reply:
Also I think we need the training of the models somewhere, as this makes the experiments reproducible. However we can't really put it in the model repository, as it relies on the classes in Carla etc.

Reply:
Ok, you have me convinced, lets continue like this then

Reply:
Should I also add the TreeModel and ForestModel from the xbgoost branch to the mlmodel catalog? Now they are inside the recourse method itself, which isn't great in my opinion.